OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -88,9 +88,10 @@ void PhysicalFrame::extendConnectToModel(Model& aModel)
 {
     Super::extendConnectToModel(aModel);
 
-    // TODO: Better use name search or more robust method
-    if (upd_geometry(0).getFrameName() == "")
-        upd_geometry(0).setFrameName(getName());
+    // 
+    int idx = getProperty_geometry().findIndexForName("frame_geometry");
+    if (idx != -1 && upd_geometry(idx).getFrameName() == "")
+        upd_geometry(idx).setFrameName(getName());
```

Comment:
This seems like a bad hack to me. Is there some well-defined convention that a Frame's geometry list will have a piece of geometry on it whose name is literally `frame_geometry` unquote and that if so, that piece of geometry should have its frame name automatically set to the name of the current Frame object, unless someone set it somewhere else? And why is this in PhysicalFrame rather than Frame? The `geometry` property itself belongs to ModelComponent. If this code is correct I think it needs a very long comment explaining why. 

@aseth1 does this seem right to you?